examples/htlc.rs

```diff
@@ -39,31 +39,34 @@ fn main() {
     )
     .expect("Resource limits");
 
-    // Check whether the descriptor is safe
-    // This checks whether all spend paths are accessible in bitcoin network.
-    // It maybe possible that some of the spend require more than 100 elements in Wsh scripts
-    // Or they contain a combination of timelock and heightlock.
+    // Check whether the descriptor is safe. This checks whether all spend paths are accessible in
+    // the Bitcoin network. It may be possible that some of the spend paths require more than 100
+    // elements in Wsh scripts or they contain a combination of timelock and heightlock.
     assert!(htlc_descriptor.sanity_check().is_ok());
```

Comment:
I would really like to get rid of these sanity checks. I think we lose of using rust if users have to call sanity check to double-check things. This feels like CPubKey.isVliad() type stuff

Reply:
Yeah, I think they don't belong in the example. They'd be useful for users who are manually constructing the `Descriptor` object I think.

Reply:
Should we be doing the sanity checks in all the constructors (incl. `FromStr`) instead?